### PR TITLE
Adding PowerPC (ppc64le) support using vector instructions for AES/GCM functions.

### PR DESCRIPTION
**Pull request by dannytsen**
The supporting functions are in PowerPC assembly for ppc64le.
  1. added aes-ecb encrypt and decrypt functions.
  2. added AES key scheduling functions.
  3. added GCM multiplication functions.

## Description

Please write a few sentences describing the overall goals of the pull request's commits.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [ ] **changelog** provided, or not required
- [ ] **backport** done, or not required
- [ ] **tests** provided, or not required

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

### CODE REVIEW

**Comment by tom-cosgrove-arm** [comment]
This would be the first `.S` file in the repo, which might cause issues for some people. All other assembler is handled with `asm` statements in `.c` files. I'm adding the `needs-design-approval` label so that we consider this

File: library/aes.c [hunk 1]
```diff
@@ -551,6 +559,10 @@ MBEDTLS_MAYBE_UNUSED static unsigned mbedtls_aes_rk_offset(uint32_t *buf)
     }
 #endif
 
+#if defined(MBEDTIS_USE_PPC)
```
[the remark this answers is not in the document]
Comment (dannytsen) [reply library/aes.c]: Fixed.

**Comment by tom-daubney-arm** [comment]
Hi @dannytsen , your latest commit is missing the sign off and hence the PR is failing the DCO checks. Please sign it off and push again, thanks.

**Comment by dannytsen** [comment]
> This would be the first `.S` file in the repo, which might cause issues for some people. All other assembler is handled with `asm` statements in `.c` files. I'm adding the `needs-design-approval` label so that we consider this

Hi @tom-daubney-arm , it's been a while.  Any info do you need from me to get this approval?  Or any suggestion?  Thanks.

**Comment by gilles-peskine-arm** [comment]
I second Tom C's point: having assembly source files complicates build scripts, so it would be better to avoid them. I would prefer to use intrinsics if recent versions of GCC and Clang have them, do they? If not then I would prefer to use inline assembly in `.c` files (in GCC-like syntax).

Also, note that the CI is failing. It looks like the build is broken with armcc/armclang (some non-portable architecture detection?) and also some style checks are failing.

**Comment by daverodgman** [comment]
Also please note that this may conflict with #8716 and probably will conflict with its' follow-up PR #8813.

**Pull request closed without merging**